api/v1alpha1/databasecluster_types.go

```diff
@@ -310,6 +311,14 @@ type Monitoring struct {
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
 }
 
+// Sharding are the sharding options. Available only for psmdb
+type Sharding struct {
+	Enabled      bool                 `json:"enabled"`
+	Shards       int32                `json:"shards"`
+	ConfigServer *psmdbv1.ReplsetSpec `json:"configServer,omitempty"`
```

Comment:
Yeah maybe we shouldn't expose the psmdbv1.ReplsetSpec. I think that right now we only need to actually change the number of config server replicas so maybe we create a
```
type ConfigServer struct {
    Replicas *int32 `json:"replicas,omitempty"`
}
```
and we expand it in the future if needed. WDYT?

Reply:
Sounds good to me, especially considering that it's the only struct we need to expose.

Reply:
Perhaps we don't even need a new object for configServer. We might use `.db.Spec.Proxy.Replicas` as we do for `pxc` and `pg`. @recharte  wdyt?

Reply:
No, ConfigServer is different from Mongos, `.db.Spec.Proxy.Replicas` should match up to `.spec.sharding.mongos.size`